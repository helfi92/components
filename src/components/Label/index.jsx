import { Component } from 'react';
import { bool, node, oneOf } from 'prop-types';
import classNames from 'classnames';
import { withStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import withTheme from '../../utils/withTheme';

@withTheme
@withStyles(theme => ({
  dense: {
    minHeight: 20,
    padding: '4px 10px 3px',
  },
  mini: {
    fontSize: '0.7rem',
    padding: '3px 8px 2px',
  },
  disabled: {
    color: 'white',
  },
  error: {
    backgroundColor: `${theme.palette.error.dark} !important`,
    color: `${theme.palette.error.contrastText} !important`,
  },
  success: {
    backgroundColor: `${theme.palette.success[700]} !important`,
    color: `${theme.palette.success.contrastText} !important`,
  },
  warning: {
    backgroundColor: `${theme.palette.warning[700]} !important`,
    color: `${theme.palette.warning.contrastText} !important`,
  },
  default: {
    backgroundColor: `${theme.palette.grey[700]} !important`,
    color: `${theme.palette.getContrastText(
      theme.palette.grey[700]
    )} !important`,
  },
  info: {
    backgroundColor: `${theme.palette.info[700]} !important`,
    color: `${theme.palette.info.contrastText} !important`,
  },
}))
/**
 * A label color-coded based on a given status.
 */
export default class Label extends Component {
  static propTypes = {
    /**
     * Content to render within the label.
     */
    children: node.isRequired,
    /**
     * An intent-driven color indicator.
     */
    status: oneOf(['error', 'success', 'warning', 'default', 'info'])
      .isRequired,
    /**
     * Show label using dense styling.
     */
    mini: bool,
  };

  static defaultProps = {
    mini: false,
  };

  render() {
    const { children, className, classes, mini, status, ...props } = this.props;

    return (
      <Button
        size="small"
        disabled
        className={classNames({
          [classes.mini]: mini,
          [className]: true,
        })}
        classes={{
          sizeSmall: classes.dense,
          disabled: classNames(classes[status], classes.disabled),
        }}
        {...props}>
        {children}
      </Button>
    );
  }
}
