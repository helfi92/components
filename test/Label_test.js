import React from 'react';
import Label from '../src/components/Label';

describe('<Label />', () => {
  it('should render correctly', () => {
    const text = 'Foo';
    const label = mount(
      <Label status="default">{text}</Label>
    );

    expect(label).toMatchSnapshot();
  });
});
