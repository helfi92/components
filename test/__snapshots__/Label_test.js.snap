// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> should render correctly 1`] = `
<WithStyles(Label)
  status="default"
>
  <Label
    className={null}
    classes={
      Object {
        "default": "Label-default-7",
        "dense": "Label-dense-1",
        "disabled": "Label-disabled-3",
        "error": "Label-error-4",
        "info": "Label-info-8",
        "mini": "Label-mini-2",
        "success": "Label-success-5",
        "warning": "Label-warning-6",
      }
    }
    mini={false}
    status="default"
  >
    <WithStyles(Button)
      className=""
      classes={
        Object {
          "disabled": "Label-default-7 Label-disabled-3",
          "sizeSmall": "Label-dense-1",
        }
      }
      disabled={true}
      size="small"
    >
      <Button
        className=""
        classes={
          Object {
            "colorInherit": "MuiButton-colorInherit-30",
            "contained": "MuiButton-contained-20",
            "containedPrimary": "MuiButton-containedPrimary-21",
            "containedSecondary": "MuiButton-containedSecondary-22",
            "disabled": "MuiButton-disabled-29 Label-default-7 Label-disabled-3",
            "extendedFab": "MuiButton-extendedFab-27",
            "fab": "MuiButton-fab-26",
            "flat": "MuiButton-flat-14",
            "flatPrimary": "MuiButton-flatPrimary-15",
            "flatSecondary": "MuiButton-flatSecondary-16",
            "focusVisible": "MuiButton-focusVisible-28",
            "fullWidth": "MuiButton-fullWidth-34",
            "label": "MuiButton-label-10",
            "mini": "MuiButton-mini-31",
            "outlined": "MuiButton-outlined-17",
            "outlinedPrimary": "MuiButton-outlinedPrimary-18",
            "outlinedSecondary": "MuiButton-outlinedSecondary-19",
            "raised": "MuiButton-raised-23",
            "raisedPrimary": "MuiButton-raisedPrimary-24",
            "raisedSecondary": "MuiButton-raisedSecondary-25",
            "root": "MuiButton-root-9",
            "sizeLarge": "MuiButton-sizeLarge-33",
            "sizeSmall": "MuiButton-sizeSmall-32 Label-dense-1",
            "text": "MuiButton-text-11",
            "textPrimary": "MuiButton-textPrimary-12",
            "textSecondary": "MuiButton-textSecondary-13",
          }
        }
        color="default"
        component="button"
        disableFocusRipple={false}
        disabled={true}
        fullWidth={false}
        mini={false}
        size="small"
        type="button"
        variant="text"
      >
        <WithStyles(ButtonBase)
          className="MuiButton-root-9 MuiButton-text-11 MuiButton-flat-14 MuiButton-sizeSmall-32 Label-dense-1 MuiButton-disabled-29 Label-default-7 Label-disabled-3"
          component="button"
          disabled={true}
          focusRipple={true}
          focusVisibleClassName="MuiButton-focusVisible-28"
          type="button"
        >
          <ButtonBase
            centerRipple={false}
            className="MuiButton-root-9 MuiButton-text-11 MuiButton-flat-14 MuiButton-sizeSmall-32 Label-dense-1 MuiButton-disabled-29 Label-default-7 Label-disabled-3"
            classes={
              Object {
                "disabled": "MuiButtonBase-disabled-36",
                "focusVisible": "MuiButtonBase-focusVisible-37",
                "root": "MuiButtonBase-root-35",
              }
            }
            component="button"
            disableRipple={false}
            disableTouchRipple={false}
            disabled={true}
            focusRipple={true}
            focusVisibleClassName="MuiButton-focusVisible-28"
            tabIndex="0"
            type="button"
          >
            <button
              className="MuiButtonBase-root-35 MuiButtonBase-disabled-36 MuiButton-root-9 MuiButton-text-11 MuiButton-flat-14 MuiButton-sizeSmall-32 Label-dense-1 MuiButton-disabled-29 Label-default-7 Label-disabled-3"
              disabled={true}
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex="-1"
              type="button"
            >
              <span
                className="MuiButton-label-10"
              >
                Foo
              </span>
            </button>
          </ButtonBase>
        </WithStyles(ButtonBase)>
      </Button>
    </WithStyles(Button)>
  </Label>
</WithStyles(Label)>
`;
